From b55895408a0fe8df3215bb693355d43a30d9d995 Mon Sep 17 00:00:00 2001
From: Vaishnav Achath <vaishnav.a@ti.com>
Date: Fri, 28 Feb 2025 10:56:34 +0530
Subject: [PATCH 24/25] PENDING: arm64: dts: ti: k3-fpdlink/v3link-* : Make
 deserializer labels generic

Rename the deserializer labels from ds90ub960_* to deserializer_*.
This helps to reuse the same camera/serializer overlays with different
deserializer such as UB960 or UB9702. Propagate the same change to all
the deserializer and camera overlays. Also while at it add ti,cdr-mode
property to all the serializer overlay to enable same overlay to work
with FPDLink 3 and 4 Deserializers.

Signed-off-by: Vaishnav Achath <vaishnav.a@ti.com>
---
 arch/arm64/boot/dts/ti/k3-am68-sk-v3link-fusion.dtso    | 8 ++++----
 arch/arm64/boot/dts/ti/k3-fpdlink-imx390-cm-0-0.dtso    | 4 ++--
 arch/arm64/boot/dts/ti/k3-fpdlink-imx390-cm-0-1.dtso    | 4 ++--
 arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-0-0.dtso   | 5 +++--
 arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-0-1.dtso   | 5 +++--
 arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-0-2.dtso   | 5 +++--
 arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-0-3.dtso   | 5 +++--
 arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-1-0.dtso   | 5 +++--
 arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-1-1.dtso   | 5 +++--
 arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-1-2.dtso   | 5 +++--
 arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-1-3.dtso   | 5 +++--
 arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-2-0.dtso   | 5 +++--
 arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-2-1.dtso   | 5 +++--
 arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-2-2.dtso   | 5 +++--
 arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-2-3.dtso   | 5 +++--
 arch/arm64/boot/dts/ti/k3-j721e-evm-fusion.dtso         | 8 ++++----
 arch/arm64/boot/dts/ti/k3-j721e-evm-ub954.dtso          | 4 ++--
 arch/arm64/boot/dts/ti/k3-j721e-sk-fpdlink-fusion.dtso  | 8 ++++----
 arch/arm64/boot/dts/ti/k3-j721s2-evm-fusion.dtso        | 8 ++++----
 arch/arm64/boot/dts/ti/k3-j721s2-evm-ub954.dtso         | 4 ++--
 arch/arm64/boot/dts/ti/k3-j722s-evm-fpdlink-fusion.dtso | 8 ++++----
 arch/arm64/boot/dts/ti/k3-j722s-evm-v3link-fusion.dtso  | 8 ++++----
 arch/arm64/boot/dts/ti/k3-v3link-imx219-0-0.dtso        | 4 ++--
 arch/arm64/boot/dts/ti/k3-v3link-imx219-0-1.dtso        | 4 ++--
 arch/arm64/boot/dts/ti/k3-v3link-imx219-0-2.dtso        | 4 ++--
 arch/arm64/boot/dts/ti/k3-v3link-imx219-0-3.dtso        | 4 ++--
 arch/arm64/boot/dts/ti/k3-v3link-imx219-1-0.dtso        | 4 ++--
 arch/arm64/boot/dts/ti/k3-v3link-imx219-1-1.dtso        | 4 ++--
 arch/arm64/boot/dts/ti/k3-v3link-imx219-1-2.dtso        | 4 ++--
 arch/arm64/boot/dts/ti/k3-v3link-imx219-1-3.dtso        | 4 ++--
 30 files changed, 84 insertions(+), 72 deletions(-)

diff --git a/arch/arm64/boot/dts/ti/k3-am68-sk-v3link-fusion.dtso b/arch/arm64/boot/dts/ti/k3-am68-sk-v3link-fusion.dtso
index 930b5749911c..9a3e7eadee44 100644
--- a/arch/arm64/boot/dts/ti/k3-am68-sk-v3link-fusion.dtso
+++ b/arch/arm64/boot/dts/ti/k3-am68-sk-v3link-fusion.dtso
@@ -38,7 +38,7 @@ deser@30 {
 
 		i2c-alias-pool = <0x4a 0x4b 0x4c 0x4d 0x4e 0x4f>;
 
-		ds90ub960_0_ports: ports {
+		deserializer_0_ports: ports {
 			#address-cells = <1>;
 			#size-cells = <0>;
 
@@ -54,7 +54,7 @@ ds90ub960_0_csi_out: endpoint {
 			};
 		};
 
-		ds90ub960_0_links: links {
+		deserializer_0_links: links {
 			#address-cells = <1>;
 			#size-cells = <0>;
 		};
@@ -74,7 +74,7 @@ deser@30 {
 
 		i2c-alias-pool = <0x5a 0x5b 0x5c 0x5d 0x5e 0x5f>;
 
-		ds90ub960_1_ports: ports {
+		deserializer_1_ports: ports {
 			#address-cells = <1>;
 			#size-cells = <0>;
 
@@ -90,7 +90,7 @@ ds90ub960_1_csi_out: endpoint {
 			};
 		};
 
-		ds90ub960_1_links: links {
+		deserializer_1_links: links {
 			#address-cells = <1>;
 			#size-cells = <0>;
 		};
diff --git a/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-cm-0-0.dtso b/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-cm-0-0.dtso
index b782373a18f0..b81153246f6e 100644
--- a/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-cm-0-0.dtso
+++ b/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-cm-0-0.dtso
@@ -9,7 +9,7 @@
 
 #include <dt-bindings/gpio/gpio.h>
 
-&ds90ub960_0_ports {
+&deserializer_0_ports {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
@@ -23,7 +23,7 @@ ub960_fpd3_1_in: endpoint {
 	};
 };
 
-&ds90ub960_0_links {
+&deserializer_0_links {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
diff --git a/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-cm-0-1.dtso b/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-cm-0-1.dtso
index dfd0c93d10de..fb9eddc290aa 100644
--- a/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-cm-0-1.dtso
+++ b/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-cm-0-1.dtso
@@ -9,7 +9,7 @@
 
 #include <dt-bindings/gpio/gpio.h>
 
-&ds90ub960_0_ports {
+&deserializer_0_ports {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
@@ -23,7 +23,7 @@ ub960_fpd3_1_in: endpoint {
 	};
 };
 
-&ds90ub960_0_links {
+&deserializer_0_links {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
diff --git a/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-0-0.dtso b/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-0-0.dtso
index 88a89eda87e0..71f354973044 100644
--- a/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-0-0.dtso
+++ b/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-0-0.dtso
@@ -9,7 +9,7 @@
 
 #include <dt-bindings/gpio/gpio.h>
 
-&ds90ub960_0_ports {
+&deserializer_0_ports {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
@@ -23,7 +23,7 @@ ub960_fpd3_1_in: endpoint {
 	};
 };
 
-&ds90ub960_0_links {
+&deserializer_0_links {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
@@ -32,6 +32,7 @@ link@0 {
 		i2c-alias = <0x44>;
 
 		ti,rx-mode = <3>;
+		ti,cdr-mode = <0>;
 
 		serializer: serializer {
 			compatible = "ti,ds90ub953-q1";
diff --git a/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-0-1.dtso b/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-0-1.dtso
index 2c6846b0fcba..44628f471054 100644
--- a/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-0-1.dtso
+++ b/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-0-1.dtso
@@ -9,7 +9,7 @@
 
 #include <dt-bindings/gpio/gpio.h>
 
-&ds90ub960_0_ports {
+&deserializer_0_ports {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
@@ -23,7 +23,7 @@ ub960_fpd3_1_in: endpoint {
 	};
 };
 
-&ds90ub960_0_links {
+&deserializer_0_links {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
@@ -32,6 +32,7 @@ link@1 {
 		i2c-alias = <0x45>;
 
 		ti,rx-mode = <3>;
+		ti,cdr-mode = <0>;
 
 		serializer: serializer {
 			compatible = "ti,ds90ub953-q1";
diff --git a/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-0-2.dtso b/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-0-2.dtso
index 9cc4c38a84f0..8fa5861aaf3b 100644
--- a/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-0-2.dtso
+++ b/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-0-2.dtso
@@ -9,7 +9,7 @@
 
 #include <dt-bindings/gpio/gpio.h>
 
-&ds90ub960_0_ports {
+&deserializer_0_ports {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
@@ -23,7 +23,7 @@ ub960_fpd3_1_in: endpoint {
 	};
 };
 
-&ds90ub960_0_links {
+&deserializer_0_links {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
@@ -32,6 +32,7 @@ link@2 {
 		i2c-alias = <0x46>;
 
 		ti,rx-mode = <3>;
+		ti,cdr-mode = <0>;
 
 		serializer: serializer {
 			compatible = "ti,ds90ub953-q1";
diff --git a/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-0-3.dtso b/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-0-3.dtso
index 46199bce5908..4bd6a9a0c484 100644
--- a/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-0-3.dtso
+++ b/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-0-3.dtso
@@ -9,7 +9,7 @@
 
 #include <dt-bindings/gpio/gpio.h>
 
-&ds90ub960_0_ports {
+&deserializer_0_ports {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
@@ -23,7 +23,7 @@ ub960_fpd3_1_in: endpoint {
 	};
 };
 
-&ds90ub960_0_links {
+&deserializer_0_links {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
@@ -32,6 +32,7 @@ link@3 {
 		i2c-alias = <0x47>;
 
 		ti,rx-mode = <3>;
+		ti,cdr-mode = <0>;
 
 		serializer: serializer {
 			compatible = "ti,ds90ub953-q1";
diff --git a/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-1-0.dtso b/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-1-0.dtso
index 57a73710ead6..196b5eab9f75 100644
--- a/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-1-0.dtso
+++ b/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-1-0.dtso
@@ -11,7 +11,7 @@
 
 #include <dt-bindings/gpio/gpio.h>
 
-&ds90ub960_1_ports {
+&deserializer_1_ports {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
@@ -25,7 +25,7 @@ ub960_fpd3_1_in: endpoint {
 	};
 };
 
-&ds90ub960_1_links {
+&deserializer_1_links {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
@@ -33,6 +33,7 @@ link@0 {
 		reg = <0>;
 		i2c-alias = <0x54>;
 
+		ti,cdr-mode = <0>;
 		ti,rx-mode = <3>;
 
 		serializer: serializer {
diff --git a/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-1-1.dtso b/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-1-1.dtso
index 17fded56c972..1c481167b69f 100644
--- a/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-1-1.dtso
+++ b/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-1-1.dtso
@@ -11,7 +11,7 @@
 
 #include <dt-bindings/gpio/gpio.h>
 
-&ds90ub960_1_ports {
+&deserializer_1_ports {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
@@ -25,7 +25,7 @@ ub960_fpd3_1_in: endpoint {
 	};
 };
 
-&ds90ub960_1_links {
+&deserializer_1_links {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
@@ -34,6 +34,7 @@ link@1 {
 		i2c-alias = <0x55>;
 
 		ti,rx-mode = <3>;
+		ti,cdr-mode = <0>;
 
 		serializer: serializer {
 			compatible = "ti,ds90ub953-q1";
diff --git a/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-1-2.dtso b/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-1-2.dtso
index 382c58c92883..466b1da701c1 100644
--- a/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-1-2.dtso
+++ b/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-1-2.dtso
@@ -11,7 +11,7 @@
 
 #include <dt-bindings/gpio/gpio.h>
 
-&ds90ub960_1_ports {
+&deserializer_1_ports {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
@@ -25,7 +25,7 @@ ub960_fpd3_1_in: endpoint {
 	};
 };
 
-&ds90ub960_1_links {
+&deserializer_1_links {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
@@ -34,6 +34,7 @@ link@2 {
 		i2c-alias = <0x56>;
 
 		ti,rx-mode = <3>;
+		ti,cdr-mode = <0>;
 
 		serializer: serializer {
 			compatible = "ti,ds90ub953-q1";
diff --git a/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-1-3.dtso b/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-1-3.dtso
index cac4c3a25075..f1ccbbfd925e 100644
--- a/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-1-3.dtso
+++ b/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-1-3.dtso
@@ -11,7 +11,7 @@
 
 #include <dt-bindings/gpio/gpio.h>
 
-&ds90ub960_1_ports {
+&deserializer_1_ports {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
@@ -25,7 +25,7 @@ ub960_fpd3_1_in: endpoint {
 	};
 };
 
-&ds90ub960_1_links {
+&deserializer_1_links {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
@@ -34,6 +34,7 @@ link@3 {
 		i2c-alias = <0x57>;
 
 		ti,rx-mode = <3>;
+		ti,cdr-mode = <0>;
 
 		serializer: serializer {
 			compatible = "ti,ds90ub953-q1";
diff --git a/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-2-0.dtso b/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-2-0.dtso
index 3fc888a0d2fd..31859fa5219e 100644
--- a/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-2-0.dtso
+++ b/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-2-0.dtso
@@ -11,7 +11,7 @@
 
 #include <dt-bindings/gpio/gpio.h>
 
-&ds90ub960_2_ports {
+&deserializer_2_ports {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
@@ -25,7 +25,7 @@ ub960_fpd3_1_in: endpoint {
 	};
 };
 
-&ds90ub960_2_links {
+&deserializer_2_links {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
@@ -34,6 +34,7 @@ link@0 {
 		i2c-alias = <0x64>;
 
 		ti,rx-mode = <3>;
+		ti,cdr-mode = <0>;
 
 		serializer: serializer {
 			compatible = "ti,ds90ub953-q1";
diff --git a/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-2-1.dtso b/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-2-1.dtso
index dc4d4ef71964..dc00af80f336 100644
--- a/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-2-1.dtso
+++ b/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-2-1.dtso
@@ -11,7 +11,7 @@
 
 #include <dt-bindings/gpio/gpio.h>
 
-&ds90ub960_2_ports {
+&deserializer_2_ports {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
@@ -25,7 +25,7 @@ ub960_fpd3_1_in: endpoint {
 	};
 };
 
-&ds90ub960_2_links {
+&deserializer_2_links {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
@@ -34,6 +34,7 @@ link@1 {
 		i2c-alias = <0x65>;
 
 		ti,rx-mode = <3>;
+		ti,cdr-mode = <0>;
 
 		serializer: serializer {
 			compatible = "ti,ds90ub953-q1";
diff --git a/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-2-2.dtso b/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-2-2.dtso
index 486b1fabe018..20451cffa90d 100644
--- a/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-2-2.dtso
+++ b/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-2-2.dtso
@@ -11,7 +11,7 @@
 
 #include <dt-bindings/gpio/gpio.h>
 
-&ds90ub960_2_ports {
+&deserializer_2_ports {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
@@ -25,7 +25,7 @@ ub960_fpd3_1_in: endpoint {
 	};
 };
 
-&ds90ub960_2_links {
+&deserializer_2_links {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
@@ -34,6 +34,7 @@ link@2 {
 		i2c-alias = <0x66>;
 
 		ti,rx-mode = <3>;
+		ti,cdr-mode = <0>;
 
 		serializer: serializer {
 			compatible = "ti,ds90ub953-q1";
diff --git a/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-2-3.dtso b/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-2-3.dtso
index 6b47107efba8..cbca4dff3bb1 100644
--- a/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-2-3.dtso
+++ b/arch/arm64/boot/dts/ti/k3-fpdlink-imx390-rcm-2-3.dtso
@@ -11,7 +11,7 @@
 
 #include <dt-bindings/gpio/gpio.h>
 
-&ds90ub960_2_ports {
+&deserializer_2_ports {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
@@ -25,7 +25,7 @@ ub960_fpd3_1_in: endpoint {
 	};
 };
 
-&ds90ub960_2_links {
+&deserializer_2_links {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
@@ -34,6 +34,7 @@ link@3 {
 		i2c-alias = <0x67>;
 
 		ti,rx-mode = <3>;
+		ti,cdr-mode = <0>;
 
 		serializer: serializer {
 			compatible = "ti,ds90ub953-q1";
diff --git a/arch/arm64/boot/dts/ti/k3-j721e-evm-fusion.dtso b/arch/arm64/boot/dts/ti/k3-j721e-evm-fusion.dtso
index f834b6d45d29..249d38081e2c 100644
--- a/arch/arm64/boot/dts/ti/k3-j721e-evm-fusion.dtso
+++ b/arch/arm64/boot/dts/ti/k3-j721e-evm-fusion.dtso
@@ -32,7 +32,7 @@ deser@3d {
 		clock-names = "refclk";
 		i2c-alias-pool = <0x4a 0x4b 0x4c 0x4d 0x4e 0x4f>;
 
-		ds90ub960_0_ports: ports {
+		deserializer_0_ports: ports {
 			#address-cells = <1>;
 			#size-cells = <0>;
 
@@ -48,7 +48,7 @@ ds90ub960_0_csi_out: endpoint {
 			};
 		};
 
-		ds90ub960_0_links: links {
+		deserializer_0_links: links {
 			#address-cells = <1>;
 			#size-cells = <0>;
 		};
@@ -61,7 +61,7 @@ deser@36 {
 		clock-names = "refclk";
 		i2c-alias-pool = <0x5a 0x5b 0x5c 0x5d 0x5e 0x5f>;
 
-		ds90ub960_1_ports: ports {
+		deserializer_1_ports: ports {
 			#address-cells = <1>;
 			#size-cells = <0>;
 
@@ -77,7 +77,7 @@ ds90ub960_1_csi_out: endpoint {
 			};
 		};
 
-		ds90ub960_1_links: links {
+		deserializer_1_links: links {
 			#address-cells = <1>;
 			#size-cells = <0>;
 		};
diff --git a/arch/arm64/boot/dts/ti/k3-j721e-evm-ub954.dtso b/arch/arm64/boot/dts/ti/k3-j721e-evm-ub954.dtso
index da5f040d5b70..527497df3e1c 100644
--- a/arch/arm64/boot/dts/ti/k3-j721e-evm-ub954.dtso
+++ b/arch/arm64/boot/dts/ti/k3-j721e-evm-ub954.dtso
@@ -32,7 +32,7 @@ deser@3d {
 		clocks = <&clk_fusion_25M_fixed>;
 		i2c-alias-pool = <0x4a 0x4b 0x4c 0x4d 0x4e 0x4f>;
 
-		ds90ub960_0_ports: ports {
+		deserializer_0_ports: ports {
 			#address-cells = <1>;
 			#size-cells = <0>;
 
@@ -48,7 +48,7 @@ ds90ub960_0_csi_out: endpoint {
 			};
 		};
 
-		ds90ub960_0_links: links {
+		deserializer_0_links: links {
 			#address-cells = <1>;
 			#size-cells = <0>;
 		};
diff --git a/arch/arm64/boot/dts/ti/k3-j721e-sk-fpdlink-fusion.dtso b/arch/arm64/boot/dts/ti/k3-j721e-sk-fpdlink-fusion.dtso
index 206932cf6745..78444ab31678 100644
--- a/arch/arm64/boot/dts/ti/k3-j721e-sk-fpdlink-fusion.dtso
+++ b/arch/arm64/boot/dts/ti/k3-j721e-sk-fpdlink-fusion.dtso
@@ -32,7 +32,7 @@ deser@3d {
 		clock-names = "refclk";
 		i2c-alias-pool = <0x4a 0x4b 0x4c 0x4d 0x4e 0x4f>;
 
-		ds90ub960_0_ports: ports {
+		deserializer_0_ports: ports {
 			#address-cells = <1>;
 			#size-cells = <0>;
 
@@ -48,7 +48,7 @@ ds90ub960_0_csi_out: endpoint {
 			};
 		};
 
-		ds90ub960_0_links: links {
+		deserializer_0_links: links {
 			#address-cells = <1>;
 			#size-cells = <0>;
 		};
@@ -61,7 +61,7 @@ deser@36 {
 		clock-names = "refclk";
 		i2c-alias-pool = <0x5a 0x5b 0x5c 0x5d 0x5e 0x5f>;
 
-		ds90ub960_1_ports: ports {
+		deserializer_1_ports: ports {
 			#address-cells = <1>;
 			#size-cells = <0>;
 
@@ -77,7 +77,7 @@ ds90ub960_1_csi_out: endpoint {
 			};
 		};
 
-		ds90ub960_1_links: links {
+		deserializer_1_links: links {
 			#address-cells = <1>;
 			#size-cells = <0>;
 		};
diff --git a/arch/arm64/boot/dts/ti/k3-j721s2-evm-fusion.dtso b/arch/arm64/boot/dts/ti/k3-j721s2-evm-fusion.dtso
index 8ea749048f99..e2844cc45fca 100644
--- a/arch/arm64/boot/dts/ti/k3-j721s2-evm-fusion.dtso
+++ b/arch/arm64/boot/dts/ti/k3-j721s2-evm-fusion.dtso
@@ -32,7 +32,7 @@ deser@3d {
 		clock-names = "refclk";
 		i2c-alias-pool = <0x4a 0x4b 0x4c 0x4d 0x4e 0x4f>;
 
-		ds90ub960_0_ports: ports {
+		deserializer_0_ports: ports {
 			#address-cells = <1>;
 			#size-cells = <0>;
 
@@ -48,7 +48,7 @@ ds90ub960_0_csi_out: endpoint {
 			};
 		};
 
-		ds90ub960_0_links: links {
+		deserializer_0_links: links {
 			#address-cells = <1>;
 			#size-cells = <0>;
 		};
@@ -61,7 +61,7 @@ deser@36 {
 		clock-names = "refclk";
 		i2c-alias-pool = <0x5a 0x5b 0x5c 0x5d 0x5e 0x5f>;
 
-		ds90ub960_1_ports: ports {
+		deserializer_1_ports: ports {
 			#address-cells = <1>;
 			#size-cells = <0>;
 
@@ -77,7 +77,7 @@ ds90ub960_1_csi_out: endpoint {
 			};
 		};
 
-		ds90ub960_1_links: links {
+		deserializer_1_links: links {
 			#address-cells = <1>;
 			#size-cells = <0>;
 		};
diff --git a/arch/arm64/boot/dts/ti/k3-j721s2-evm-ub954.dtso b/arch/arm64/boot/dts/ti/k3-j721s2-evm-ub954.dtso
index 9cb168132e7b..76e311af712c 100644
--- a/arch/arm64/boot/dts/ti/k3-j721s2-evm-ub954.dtso
+++ b/arch/arm64/boot/dts/ti/k3-j721s2-evm-ub954.dtso
@@ -32,7 +32,7 @@ deser@3d {
 		clocks = <&clk_fusion_25M_fixed>;
 		i2c-alias-pool = <0x4a 0x4b 0x4c 0x4d 0x4e 0x4f>;
 
-		ds90ub960_0_ports: ports {
+		deserializer_0_ports: ports {
 			#address-cells = <1>;
 			#size-cells = <0>;
 
@@ -48,7 +48,7 @@ ds90ub960_0_csi_out: endpoint {
 			};
 		};
 
-		ds90ub960_0_links: links {
+		deserializer_0_links: links {
 			#address-cells = <1>;
 			#size-cells = <0>;
 		};
diff --git a/arch/arm64/boot/dts/ti/k3-j722s-evm-fpdlink-fusion.dtso b/arch/arm64/boot/dts/ti/k3-j722s-evm-fpdlink-fusion.dtso
index 7a9b23c80702..500984a6c8b2 100644
--- a/arch/arm64/boot/dts/ti/k3-j722s-evm-fpdlink-fusion.dtso
+++ b/arch/arm64/boot/dts/ti/k3-j722s-evm-fpdlink-fusion.dtso
@@ -37,7 +37,7 @@ deser@3d {
 			clock-names = "refclk";
 			i2c-alias-pool = <0x4a 0x4b 0x4c 0x4d 0x4e 0x4f>;
 
-			ds90ub960_0_ports: ports {
+			deserializer_0_ports: ports {
 				#address-cells = <1>;
 				#size-cells = <0>;
 
@@ -53,7 +53,7 @@ ds90ub960_0_csi_out: endpoint {
 				};
 			};
 
-			ds90ub960_0_links: links {
+			deserializer_0_links: links {
 				#address-cells = <1>;
 				#size-cells = <0>;
 			};
@@ -66,7 +66,7 @@ deser@36 {
 			clock-names = "refclk";
 			i2c-alias-pool = <0x5a 0x5b 0x5c 0x5d 0x5e 0x5f>;
 
-			ds90ub960_1_ports: ports {
+			deserializer_1_ports: ports {
 				#address-cells = <1>;
 				#size-cells = <0>;
 
@@ -82,7 +82,7 @@ ds90ub960_1_csi_out: endpoint {
 				};
 			};
 
-			ds90ub960_1_links: links {
+			deserializer_1_links: links {
 				#address-cells = <1>;
 				#size-cells = <0>;
 			};
diff --git a/arch/arm64/boot/dts/ti/k3-j722s-evm-v3link-fusion.dtso b/arch/arm64/boot/dts/ti/k3-j722s-evm-v3link-fusion.dtso
index 614198ca334c..f5f55113a34a 100644
--- a/arch/arm64/boot/dts/ti/k3-j722s-evm-v3link-fusion.dtso
+++ b/arch/arm64/boot/dts/ti/k3-j722s-evm-v3link-fusion.dtso
@@ -56,7 +56,7 @@ deser@30 {
 
 			i2c-alias-pool = <0x4a 0x4b 0x4c 0x4d 0x4e 0x4f>;
 
-			ds90ub960_0_ports: ports {
+			deserializer_0_ports: ports {
 				#address-cells = <1>;
 				#size-cells = <0>;
 
@@ -72,7 +72,7 @@ ds90ub960_0_csi_out: endpoint {
 				};
 			};
 
-			ds90ub960_0_links: links {
+			deserializer_0_links: links {
 				#address-cells = <1>;
 				#size-cells = <0>;
 			};
@@ -94,7 +94,7 @@ deser@30 {
 
 			i2c-alias-pool = <0x5a 0x5b 0x5c 0x5d 0x5e 0x5f>;
 
-			ds90ub960_1_ports: ports {
+			deserializer_1_ports: ports {
 				#address-cells = <1>;
 				#size-cells = <0>;
 
@@ -110,7 +110,7 @@ ds90ub960_1_csi_out: endpoint {
 				};
 			};
 
-			ds90ub960_1_links: links {
+			deserializer_1_links: links {
 				#address-cells = <1>;
 				#size-cells = <0>;
 			};
diff --git a/arch/arm64/boot/dts/ti/k3-v3link-imx219-0-0.dtso b/arch/arm64/boot/dts/ti/k3-v3link-imx219-0-0.dtso
index 90390291bd71..d6758ce4a660 100644
--- a/arch/arm64/boot/dts/ti/k3-v3link-imx219-0-0.dtso
+++ b/arch/arm64/boot/dts/ti/k3-v3link-imx219-0-0.dtso
@@ -19,7 +19,7 @@ clk_imx219_fixed_00: imx219-xclk-00 {
 	};
 };
 
-&ds90ub960_0_ports {
+&deserializer_0_ports {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
@@ -33,7 +33,7 @@ ub960_fpd3_1_in: endpoint {
 	};
 };
 
-&ds90ub960_0_links {
+&deserializer_0_links {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
diff --git a/arch/arm64/boot/dts/ti/k3-v3link-imx219-0-1.dtso b/arch/arm64/boot/dts/ti/k3-v3link-imx219-0-1.dtso
index 9971246acb0e..4bbb546606eb 100644
--- a/arch/arm64/boot/dts/ti/k3-v3link-imx219-0-1.dtso
+++ b/arch/arm64/boot/dts/ti/k3-v3link-imx219-0-1.dtso
@@ -19,7 +19,7 @@ clk_imx219_fixed_01: imx219-xclk-01 {
 	};
 };
 
-&ds90ub960_0_ports {
+&deserializer_0_ports {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
@@ -33,7 +33,7 @@ ub960_fpd3_1_in: endpoint {
 	};
 };
 
-&ds90ub960_0_links {
+&deserializer_0_links {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
diff --git a/arch/arm64/boot/dts/ti/k3-v3link-imx219-0-2.dtso b/arch/arm64/boot/dts/ti/k3-v3link-imx219-0-2.dtso
index 8fb755f87cec..be2a8df870de 100644
--- a/arch/arm64/boot/dts/ti/k3-v3link-imx219-0-2.dtso
+++ b/arch/arm64/boot/dts/ti/k3-v3link-imx219-0-2.dtso
@@ -19,7 +19,7 @@ clk_imx219_fixed_02: imx219-xclk-02 {
 	};
 };
 
-&ds90ub960_0_ports {
+&deserializer_0_ports {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
@@ -33,7 +33,7 @@ ub960_fpd3_1_in: endpoint {
 	};
 };
 
-&ds90ub960_0_links {
+&deserializer_0_links {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
diff --git a/arch/arm64/boot/dts/ti/k3-v3link-imx219-0-3.dtso b/arch/arm64/boot/dts/ti/k3-v3link-imx219-0-3.dtso
index 25c55f0840ec..94f1aaf10ef0 100644
--- a/arch/arm64/boot/dts/ti/k3-v3link-imx219-0-3.dtso
+++ b/arch/arm64/boot/dts/ti/k3-v3link-imx219-0-3.dtso
@@ -19,7 +19,7 @@ clk_imx219_fixed_03: imx219-xclk-03 {
 	};
 };
 
-&ds90ub960_0_ports {
+&deserializer_0_ports {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
@@ -33,7 +33,7 @@ ub960_fpd3_1_in: endpoint {
 	};
 };
 
-&ds90ub960_0_links {
+&deserializer_0_links {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
diff --git a/arch/arm64/boot/dts/ti/k3-v3link-imx219-1-0.dtso b/arch/arm64/boot/dts/ti/k3-v3link-imx219-1-0.dtso
index 3d0abe4b84c4..33f7d242346f 100644
--- a/arch/arm64/boot/dts/ti/k3-v3link-imx219-1-0.dtso
+++ b/arch/arm64/boot/dts/ti/k3-v3link-imx219-1-0.dtso
@@ -19,7 +19,7 @@ clk_imx219_fixed_10: imx219-xclk-10 {
 	};
 };
 
-&ds90ub960_1_ports {
+&deserializer_1_ports {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
@@ -33,7 +33,7 @@ ub960_fpd3_1_in: endpoint {
 	};
 };
 
-&ds90ub960_1_links {
+&deserializer_1_links {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
diff --git a/arch/arm64/boot/dts/ti/k3-v3link-imx219-1-1.dtso b/arch/arm64/boot/dts/ti/k3-v3link-imx219-1-1.dtso
index 986cdb4fc0c2..cef0e5dec8d6 100644
--- a/arch/arm64/boot/dts/ti/k3-v3link-imx219-1-1.dtso
+++ b/arch/arm64/boot/dts/ti/k3-v3link-imx219-1-1.dtso
@@ -19,7 +19,7 @@ clk_imx219_fixed_11: imx219-xclk-11 {
 	};
 };
 
-&ds90ub960_1_ports {
+&deserializer_1_ports {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
@@ -33,7 +33,7 @@ ub960_fpd3_1_in: endpoint {
 	};
 };
 
-&ds90ub960_1_links {
+&deserializer_1_links {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
diff --git a/arch/arm64/boot/dts/ti/k3-v3link-imx219-1-2.dtso b/arch/arm64/boot/dts/ti/k3-v3link-imx219-1-2.dtso
index ce48f06d550a..fad2ea2a0641 100644
--- a/arch/arm64/boot/dts/ti/k3-v3link-imx219-1-2.dtso
+++ b/arch/arm64/boot/dts/ti/k3-v3link-imx219-1-2.dtso
@@ -19,7 +19,7 @@ clk_imx219_fixed_12: imx219-xclk-12 {
 	};
 };
 
-&ds90ub960_1_ports {
+&deserializer_1_ports {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
@@ -33,7 +33,7 @@ ub960_fpd3_1_in: endpoint {
 	};
 };
 
-&ds90ub960_1_links {
+&deserializer_1_links {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
diff --git a/arch/arm64/boot/dts/ti/k3-v3link-imx219-1-3.dtso b/arch/arm64/boot/dts/ti/k3-v3link-imx219-1-3.dtso
index 28916c4c0939..574554f7353d 100644
--- a/arch/arm64/boot/dts/ti/k3-v3link-imx219-1-3.dtso
+++ b/arch/arm64/boot/dts/ti/k3-v3link-imx219-1-3.dtso
@@ -19,7 +19,7 @@ clk_imx219_fixed_13: imx219-xclk-13 {
 	};
 };
 
-&ds90ub960_1_ports {
+&deserializer_1_ports {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
@@ -33,7 +33,7 @@ ub960_fpd3_1_in: endpoint {
 	};
 };
 
-&ds90ub960_1_links {
+&deserializer_1_links {
 	#address-cells = <1>;
 	#size-cells = <0>;
 
-- 
2.47.2

